src/client/communicator.go

```diff
@@ -64,41 +65,53 @@ func (this *Communicator) GetVmsData() (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting vms data error: %s", err)
 	}
-	s, err := this.GetRegistryVms()
-	if err != nil {
-		return nil, fmt.Errorf("getting vms registry data error: %s", err)
-	}
-	m := d.([]types.Instance)
-	for k, v := range m {
-		i := v.Vm.TemplateUUID
-		m[k].Vm.TemplateNAME = s[i]
+	templatesMap := state.GetState().GetTemplatesMap()
+	instances := d.([]types.Instance)
+	for i, v := range instances {
+		template, ok := templatesMap[v.Vm.TemplateUUID]
+		if !ok {
+			return nil, fmt.Errorf("couldn't find template name being used by instance")
```

Comment:
This is a temporary issue, so I would not return at this error but rather skip this template. you can print a warning or something.

Reply:
Added continue